pkg/server/tls_windows.go

```diff
@@ -0,0 +1,342 @@
+		subjectRegEx, err = regexp.Compile(c.WindowsCertificateFilter.ClientSubjectRegEx)
+		if err != nil {
+			return fmt.Errorf("error compiling subject common name regular expression %w", err)
+		}
+	}
+
+	// If there is an existing windows certhandler notify it to stop refreshing
+	if l.windowsCertHandler != nil {
+		l.windowsCertHandler.stopUpdateTimer <- struct{}{}
+	}
+
+	cn := &winCertStoreHandler{
+		cfg:             *c.WindowsCertificateFilter,
+		subjectRegEx:    subjectRegEx,
+		stopUpdateTimer: make(chan struct{}, 1),
+	}
+	err = cn.refreshCerts()
+	if err != nil {
+		return err
+	}
+	certPool := x509.NewCertPool()
+	certPool.AddCert(cn.clientRootCA)
+	config := &tls.Config{
+		VerifyPeerCertificate: cn.VerifyPeer,
+		GetCertificate: func(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
+			cn.winMut.Lock()
+			defer cn.winMut.Unlock()
+			cert := &tls.Certificate{
+				Certificate: [][]byte{cn.serverCert.Raw},
+				PrivateKey:  cn.serverSigner,
+				// These seem to the be safest to use, tested on Win10, Server 2016, 2019, 2022
+				SupportedSignatureAlgorithms: []tls.SignatureScheme{
+					tls.PKCS1WithSHA256,
+					tls.PKCS1WithSHA384,
+					tls.PKCS1WithSHA512,
+				},
+			}
+			return cert, nil
+		},
+		// Windows has broad support for 1.2, only 2022 has support for 1.3
+		MaxVersion: tls.VersionTLS12,
+	}
+
+	switch c.ClientAuth {
+	case "RequestClientCert":
+		config.ClientAuth = tls.RequestClientCert
+	case "RequireAnyClientCert", "RequireClientCert": // Preserved for backwards compatibility.
+		config.ClientAuth = tls.RequireAnyClientCert
+	case "VerifyClientCertIfGiven":
+		config.ClientAuth = tls.VerifyClientCertIfGiven
+	case "RequireAndVerifyClientCert":
+		config.ClientAuth = tls.RequireAndVerifyClientCert
+	case "", "NoClientCert":
+		config.ClientAuth = tls.NoClientCert
+	default:
+		return fmt.Errorf("invalid ClientAuth %q", c.ClientAuth)
+	}
+	// Kick off the refresh handler
+	go cn.startUpdateTimer()
+	l.windowsCertHandler = cn
+	l.tlsConfig = config
+	l.cfg = c
+	return nil
+}
+
+// this is the ASN1 Object Identifier for TemplateID
+var asnTemplateOID = "1.3.6.1.4.1.311.21.7"
+
+type winCertStoreHandler struct {
+	cfg          WindowsCertificateFilter
+	subjectRegEx *regexp.Regexp
+
+	winMut       sync.Mutex
+	serverCert   *x509.Certificate
+	serverSigner crypto.PrivateKey
+	// We have to store the identity to access the signer (its an api call), the client does NOT need the signer
+	serverIdentity certstore.Identity
+	clientRootCA   *x509.Certificate
+
+	stopUpdateTimer chan struct{}
+}
+
+func (c *winCertStoreHandler) startUpdateTimer() {
+	if c.cfg.ServerRefreshInterval == 0 {
+		c.cfg.ServerRefreshInterval = 5 * time.Minute
+	}
+
+	select {
+	case <-c.stopUpdateTimer:
+		if c.serverIdentity != nil {
+			c.serverIdentity.Close()
+		}
+		c.serverCert = nil
+		c.serverSigner = nil
+		return
+	case <-time.After(c.cfg.ServerRefreshInterval):
+		err := c.refreshCerts()
+		if err != nil {
+			return
+		}
```

Comment:
Two things:

1. This should be logging the error 
2. Is returning expected? Or should we try again later?

Reply:
Agreed, and with fixing the for loop it will try again at the interval. In the case of an error in refreshCerts the previous certificates will be used.